src/PetProject.Framework.Kafka/Consumer/Consumer.cs

```diff
@@ -91,65 +88,73 @@ public void Dispose()
         /// <summary>
         /// Decorator around Confluent Consumer to commit messages asynchronously after success.
         /// </summary>
-        public void CommitAsync()
+        public async Task<CommittedOffsets> CommitAsync()
         {
-            this.confluentConsumer.CommitAsync();
+            return await this.confluentConsumer.CommitAsync();
         }
 
         /// <summary>
         /// Method to add custom treatment to Consumer Statistics.
         /// </summary>
         /// <param name="sender">Message sender.</param>
         /// <param name="statistics">Statistics returned from Confluent Consumer.</param>
-        protected abstract void HandleStatistics(object sender, string statistics);
+        protected virtual void HandleStatistics(object sender, string statistics)
+        {
+            this.logger.LogInformation("Framework Kafka Statistics: {statistics}", statistics);
+        }
 
         /// <summary>
         /// Method to add custom treatment to Consumer Logs.
         /// </summary>
         /// <param name="sender">Message sender.</param>
         /// <param name="logMessage">Log Message object returned from Confluent Consumer.</param>
-        protected abstract void HandleLogs(object sender, LogMessage logMessage);
+        protected virtual void HandleLogs(object sender, LogMessage logMessage)
+        {
+            this.logger.LogInformation("Framework Kafka LogMessage: {logMessage}", logMessage);
+        }
 
         /// <summary>
         /// Method to add custom treatment to Consumer Errors.
         /// </summary>
         /// <param name="sender">Message sender</param>
         /// <param name="error">Error object returned from Confluent Consumer</param>
-        protected abstract void HandleError(object sender, Error error);
+        protected virtual void HandleError(object sender, Error error)
+        {
+            this.logger.LogError("Framework Kafka Error: {error}", error);
+        }
 
         /// <summary>
         /// Must be implemented by the client in order to add Errors while consuming.
         /// </summary>
         /// <param name="sender">Message sender.</param>
         /// <param name="message">Message envelope with the content to consume.</param>
-        protected abstract void HandleOnConsumerError(object sender, Confluent.Kafka.Message message);
+        protected virtual void HandleOnConsumerError(object sender, Message message)
+        {
+            this.logger.LogWarning("Framework Kafka ConsumerError: {message}", message);
+            this.RequeueMessageOnError(message);
+        }
+
+        protected abstract void RequeueMessageOnError(Message message);
```

Comment:
we should provide a default implementation of this method, 99% of the times it will be the same, it's just producing the same message to the same topic.

Reply:
ok